### PR TITLE
SimplePerf updates

### PR DESCRIPTION
**Pull request by bill-long**
* Breaking change: -IncludeThread switch is gone. Instead, pass -IncludeCounters "\Thread".
* Now supports two scenarios, Exchange and None. Easy to add more.
* Now supports -IncludeCounters and -ExcludeCounters to customize counter sets.
* Now supports naming collectors, so you can run more than one SimplePerf at a time. This feature is currently not in the docs because I ran out of time.
* Exchange scenario is slimmed down and should result in less data collected than before - closer to the ExPerfWiz defaults.

### CODE REVIEW

**Review by lusassl-msft — COMMENTED** [review]
Change looks good and seems to work (validated in lab). However, the default exclude/include doesn't work properly on systems which are using a different language than en-US. We developed the `PerformanceCountersFunctions.ps1` helper function which we use in `HealthChecker` to address this kind of problem. It might be a good idea to use this helper here as well. 

image


See #875

**Comment by dpaulson45** [comment]
> Change looks good and seems to work (validated in lab). However, the default exclude/include doesn't work properly on systems which are using a language other than `en-US`. We've developed the `PerformanceCountersFunctions.ps1` helper function which we use in `HealthChecker` to address this kind of problem. It might be a good idea to use this helper here as well.
> 
> image
> 
> See #875

The filters should still work if you provide the language specific counters for that system.

**Comment by bill-long** [comment]
The latest commit translates the counter names. Note that this means you must always enter counter names in English, even if your current language is _not_ English:

image

However, one problem I haven't figured out how to fix:

image

The script tries to determine if a collector is already running, etc, but those status names are also localized. Need to figure out a language-agnostic way of determining if a collector is running.

**Comment by bill-long** [comment]
Got it:

image

**Comment by lusassl-msft** [comment]
Can confirm that it works as expected on German OS. 
We should document this in the docs:

_[...] Note that this means you must always enter counter names in English, even if your current language is not English_

**Review by lusassl-msft — DISMISSED** [review]

File: Performance/SimplePerf/SimplePerf.ps1 [hunk 1]
```diff
@@ -236,7 +240,10 @@ begin {
 
         $counterFullNames = $countersFiltered | ForEach-Object { ("\\localhost" + $_) }
 
-        $counterFullNames | ForEach-Object { Write-Verbose $_ }
+        if ($DisplayFilterResults) {
+            Write-Host "$($env:COMPUTERNAME): The following counters matched the specified filters:"
```
[the remark this answers is not in the document]
Comment (bill-long) [reply Performance/SimplePerf/SimplePerf.ps1:244]: Ugh, well I just redid all the screenshots. That will have to wait for a future PR.

[the remark this answers is not in the document]
Comment (bill-long) [reply Performance/SimplePerf/SimplePerf.ps1:244]: Also it doesn't work:

image

File: Performance/SimplePerf/SimplePerf.ps1 [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        $CollectorName,
+        $($VerbosePreference -ne 'SilentlyContinue')
+    )
+
+    if ($computerTargets.Length -gt 0) {
+        foreach ($computer in $computerTargets) {
+            if ($Start) {
+                Invoke-Command -ComputerName $computer -ScriptBlock ${function:StartSimplePerf} -ArgumentList $argumentList
+            } elseif ($Stop) {
+                Invoke-Command -ComputerName $computer -ScriptBlock ${function:StopSimplePerf}
```
Comment (dpaulson45) [inline Performance/SimplePerf/SimplePerf.ps1:315]: This isn't going to work if a different `$CollectorName` is used.

**Review by dpaulson45 — CHANGES_REQUESTED** [review]

File: Performance/SimplePerf/SimplePerf.ps1 (hunk 2 above)
Comment (bill-long) [reply Performance/SimplePerf/SimplePerf.ps1]: Fixed.

image

**Review by dpaulson45 — APPROVED** [review]

**Pull request merged**